File: examples/ruby/sinatra/reject.rb
require 'sinatra'
require 'builder'

set :port, 3000

post '/' do
  logger.info "from: #{params[:from]}"
  logger.info "to: #{params[:to]}"
  logger.info "direction: #{params[:direction]}"

  headers 'Content-Type' => "application/xml"
  xml = Builder::XmlMarkup.new(:indent => 2)
  xml.instruct!
  xml.Response{xml.Reject("reason" => "busy")}
# xml.Response{xml.Reject}

end

File: examples/nodejs/express/reject.js
var app = require('express')();
var bodyParser = require('body-parser');
var xml = require('xml');

app.use(bodyParser.urlencoded({ extended: false }));

app.post("/", function (request, response) {
	var from = request.body.from;
	var to = request.body.to;
	var direction = request.body.direction;

	console.log("from: " + from);
	console.log("to: " + to);
	console.log("direction: " + direction);

	response.set('Content-Type', 'application/xml');
	response.send(
		xml({ Response: [
			{ Reject: { _attr: { reason: "busy" } } },
			// { Reject: null },
		] })
	);
});

app.listen(3000);
